tests/transform/test_gene_info.py

```diff
@@ -0,0 +1,303 @@
+        "median_expression": "median_expression_good_input.csv",
+        "druggability": "druggability_good_input.csv",
+        "genes_biodomains": "genes_biodomains_good_input.csv",
+        "tep_adi_info": "tep_adi_info_good_input.csv",
+    }
+
+    pval_error_match_string = "'<=' not supported"
+    merge_error_match_string = "Merge keys are not unique"
+
+    pass_test_data = [
+        (  # Pass with good data on param set 1
+            core_files,
+            "gene_info_good_output_1.json",
+            param_set_1,
+        ),
+        (  # Pass with good data on param set 2
+            core_files,
+            "gene_info_good_output_2.json",
+            param_set_2,
+        ),
+    ]
+    pass_test_ids = [
+        "Pass with good data on parameter set 1",
+        "Pass with good data on parameter set 2",
+    ]
+    fail_test_data = [
+        (  # Duplicate Ensembl IDs in gene_metadata
+            core_files,
+            {"gene_metadata": "gene_metadata_merge_error.feather"},
+            param_set_1,
+            pd.errors.MergeError,
+            merge_error_match_string,
+        ),
+        (  # Duplicate Ensembl IDs in igap
+            core_files,
+            {"igap": "igap_merge_error.csv"},
+            param_set_1,
+            pd.errors.MergeError,
+            merge_error_match_string,
+        ),
+        (  # Duplicate Ensembl IDs in eqtl
+            core_files,
+            {"eqtl": "eqtl_merge_error.csv"},
+            param_set_1,
+            pd.errors.MergeError,
+            merge_error_match_string,
+        ),
+        (  # Bad data type in diff_exp_data
+            core_files,
+            {"diff_exp_data": "diff_exp_data_type_error.csv"},
+            param_set_1,
+            TypeError,
+            pval_error_match_string,
+        ),
+        (  # Bad data type in proteomics
+            core_files,
+            {"proteomics": "proteomics_type_error.csv"},
+            param_set_1,
+            TypeError,
+            pval_error_match_string,
+        ),
+        (  # Bad data type in proteomics_tmt
+            core_files,
+            {"proteomics_tmt": "proteomics_tmt_type_error.csv"},
+            param_set_1,
+            TypeError,
+            pval_error_match_string,
+        ),
+        (  # Bad data type in proteomics_srm
+            core_files,
+            {"proteomics_srm": "proteomics_srm_type_error.csv"},
+            param_set_1,
+            TypeError,
+            pval_error_match_string,
+        ),
+        (  # Missing HGNC in tep_adi_info
+            core_files,
+            {"tep_adi_info": "tep_adi_info_type_error_1.csv"},
+            param_set_1,
+            TypeError,
+            "can only concatenate str",
+        ),
+        (  # is_adi is a string
+            core_files,
+            {"tep_adi_info": "tep_adi_info_type_error_2.csv"},
+            param_set_1,
+            TypeError,
+            "wrong data type in 'is_adi' column",
```

Comment:
Just a note that if you update the error messages from my other comment you will need to update these too